### PR TITLE
Migrating DateFunctionsTest, DeleteTest

### PR DESCRIPTION
**Pull request by abbashus**
*Issue #17 :*

*Testing: * Passing gradle build.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DateFunctionsIT.java [hunk 1]
```diff
@@ -201,19 +207,18 @@ public void secondOfMinute() {
...
 
-    private SearchHit[] execute(String sql) {
-        try {
-            SearchDao searchDao = MainTestSuite.getSearchDao();
-            SqlElasticSearchRequestBuilder select = (SqlElasticSearchRequestBuilder) searchDao.explain(sql).explain();
-            return ((SearchResponse) select.get()).getHits().getHits();
-        } catch (SQLFeatureNotSupportedException | SqlParseException e) {
-            throw new ParserException("Unable to parse query: " + sql);
-        }
+    private SearchHit[] execute(String sqlRequest) throws IOException {
```
Comment (arsen-es) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DateFunctionsIT.java]: Not a blocker: I think this code is now re-used in multiple classes, would be good to move to the base class.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        JSONObject response = executeRequest(makeRequest(selectQuery));
+        int totalHits = getTotalHits(response);
+
+        String deleteQuery = String.format(
+            Locale.ROOT,
+            "DELETE FROM %s/account",
+            TestsConstants.TEST_INDEX_ACCOUNT);
+        response = executeRequest(makeRequest(deleteQuery));
+        assertThat(response.getInt("deleted"), equalTo(totalHits));
+        Thread.sleep(2000);
```
Comment (arsen-es) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java]: can you add a comment to explain why we need to do `sleep`

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        int totalHits = getTotalHits(response);
+
+        String deleteQuery = String.format(
+            Locale.ROOT,
+            "DELETE FROM %s/phrase WHERE match_phrase(phrase, 'quick fox here')",
+            TestsConstants.TEST_INDEX_PHRASE
+        );
+        response = executeRequest(makeRequest(deleteQuery));
+        assertThat(response.getInt("deleted"), equalTo(totalHits));
+        Thread.sleep(2000);
```
Comment (arsen-es) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java]: same here

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DateFunctionsIT.java (hunk 1 above)
Comment (abbashus) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DateFunctionsIT.java]: Sure, will take this in the next commit. Looks like it needs a lot of refactoring. Will mark it with `TODO`.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java (hunk 3 above)
Comment (abbashus) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java]: Sure, will do.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java (hunk 2 above)
Comment (abbashus) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DeleteIT.java]: Sure, will do.

**Pull request merged**